### ISSUE TITLE
Making it easy to add comments

### ISSUE THREAD

**Issue by dplanella** [gollum/gollum#1222]
It is relatively easy to add Disqus comments to a Gollum wiki by using their [universal embed code](https://davidplanella.disqus.com/admin/universalcode/).

That works fine for the most basic case, but it messes up the commenting system when pages are renamed in Gollum, as there is no way to track a page by a unique identifier. That is, at the time of writing there is no way to specify a meaningful value for these variables in the Disqus JS embed code snippet:

```
this.page.url = PAGE_URL;  // Replace PAGE_URL with your page's canonical URL variable
this.page.identifier = PAGE_IDENTIFIER; // Replace PAGE_IDENTIFIER with your page's unique id
``` 

The way I added Disqus comments was to simply add a mustache partial template (`{{>comments}}`)  that contains the Disqus embed code, and then modified the stock `page.template` to reference it. Finally, I copied all stock templates plus the new `comments.mustache` and modified `page.mustache` ones to a location of my choice and pointed Gollum to it via the `template-dir` option (see also issue #1221).

Using that method, and unless I'm mistaken, limits me to use only the Ruby variables defined in page.rb in the page.mustache and included partial templates.

Would it be possible to expose a unique page ID and the page URL there? Happy to be pointed to an alternative way to access them, too.

Thanks!

**Comment by dometto** [comment]
There is not really such a thing as a unique identifier of a file in git. Doing `git mv` basically amounts to deleting the file at the old path, and recreating a file with the same contents at a different path. The closest thing is the page's `SHA1` hash -- but of course it's possible to have different pages with the same content and so the same hash in your repo. I would be happy to expose a method returning the page's `SHA1` hash though, if that works for your.

Alternatively, perhaps you could fabricate unique ids yourself, and store them in the page as metadata tags. Use the following syntax:
`<!-- --- unique_id: 31415 -->`
 Metadata is available in the `Page` view, so you should be able to reference it in the template.

Note that starting from version `5.0`, gollum will use [YAML front matter](https://github.com/gollum/gollum/wiki/5.0-release-notes#metadata-yaml-frontmatter) instead of these old-style metadata tags.

**Comment by dplanella** [comment]
Thanks a lot for the reply. I had actually thought about the metadata too, but I discarded it as it would be a pain to maintain a manual index for each page. I noticed the upcoming change of metadada format, thanks for the tip and for the release notes. Looking forward to 5.0!

I think exposing the SHA1 hash would be quite useful for this purpose. I understand that it's not unique as you are saying, but I think you'd generally not have two pages with the exact same content on a wiki.
